p2p/utils_test.go

```diff
@@ -0,0 +1,65 @@
+package p2p
+
+import (
+	"context"
+	"testing"
+
+	mocknet "github.com/libp2p/go-libp2p/p2p/net/mock"
+	"github.com/stretchr/testify/require"
+	"go.uber.org/multierr"
+
+	"github.com/lazyledger/optimint/config"
+	"github.com/lazyledger/optimint/log"
+)
+
+type testNet []*Client
+
+func (tn testNet) Close() (err error) {
+	for i := range tn {
+		err = multierr.Append(err, tn[i].Close())
+	}
+	return
+}
+
+func (tn testNet) WaitForDHT() {
+	for i := range tn {
+		<-tn[i].dht.RefreshRoutingTable()
+	}
+}
+
+func startTestNetwork(t *testing.T, n int, conns map[int][]int, logger log.Logger) testNet {
+	t.Helper()
+	require := require.New(t)
+
+	mnet, err := mocknet.FullMeshLinked(context.Background(), n)
```

Comment:
Better create context at the beginning of the context and pass it here. That might be helpful in some tests